pkg/oc/admin/diagnostics/diagnostics/cluster/scc.go

```diff
@@ -0,0 +1,95 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	kerrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/authorization"
+	authorizationtypedclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/authorization/internalversion"
+
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/types"
+	"github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/util"
+	policycmd "github.com/openshift/origin/pkg/oc/admin/policy"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	securitytypedclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+)
+
+// ClusterSCC is a Diagnostic to check that the default securitycontextconstraints match expectations
+type ClusterSCC struct {
+	SCCClient securitytypedclient.SecurityContextConstraintsInterface
+	SARClient authorizationtypedclient.SelfSubjectAccessReviewsGetter
+}
+
+const SccName = "ClusterSCC"
+
+func (d *ClusterSCC) Name() string {
+	return SccName
+}
+
+func (d *ClusterSCC) Description() string {
+	return "Check that the default SecurityContextConstraints are present and contain the expected permissions"
+}
+
+func (d *ClusterSCC) Requirements() (client bool, host bool) {
+	return true, false
+}
+
+func (d *ClusterSCC) CanRun() (bool, error) {
+	if d.SCCClient == nil {
+		return false, fmt.Errorf("must have client.SubjectAccessReviews")
+	}
+
+	return util.UserCan(d.SARClient, &authorization.ResourceAttributes{
+		Verb:     "list",
+		Group:    securityapi.GroupName,
+		Resource: "securitycontextconstraints",
+	})
+}
+
+func (d *ClusterSCC) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(SccName)
+	reconcileOptions := &policycmd.ReconcileSCCOptions{
+		Confirmed:      false,
+		Union:          true,
+		Out:            ioutil.Discard,
+		SCCClient:      d.SCCClient,
+		InfraNamespace: bootstrappolicy.DefaultOpenShiftInfraNamespace,
+	}
+
+	changedSCCs, err := reconcileOptions.ChangedSCCs()
+	if err != nil {
+		r.Error("CSD1000", err, fmt.Sprintf("Error inspecting SCCs: %v", err))
+		return r
+	}
+	changedSCCNames := map[string]bool{}
+	for _, changedSCC := range changedSCCs {
+		_, err := d.SCCClient.Get(changedSCC.Name, metav1.GetOptions{})
```

Comment:
It looks like `ChangedSCCs()` could be more useful if it would return 2 different sets of SCCs -- one for non-existing and another one that need to be updated. Perhaps, it can be implemented as a separate PR..

Reply:
Yes, that's useful. But it would be great if it is implemented as a separate PR.

Reply:
@stlaz Would you be interested in implementing such a refactoring?

Reply:
Sure!

Reply:
I've had the refactoring ready for quite some time now. Given that we're waiting for the rebase, should I just post it to this PR or transitively wait for the rebase to happen so that this PR is merged and create a new PR with it afterwards?

Reply:
I'd rather base your PR on top of this one.